ynput/ayon-core
# Max: TyCache enhancement cached material

## Changelog Description
Related to https://github.com/ynput/ayon-core/pull/56
This PR is to allow AYON to publish the material(s) assigned to tyCache along with all cached data.

## Additional info
Simple Remark on the enhancement in the future, we can support instanced material override in the loader after the implementation of look family (i.e. the publish of the shader in Max)

## Testing notes:
1. Launch Max
2. Open the scene with tyCache 
3. Assign some materials to tyCache by adding instance material nodes
4. Use any scene materials for the tyCache node
![image](https://github.com/ynput/ayon-core/assets/64118225/47fb1ca6-dc7f-46bf-8198-a5b68504876a)

5. Create tyCache instance with the tyCache node
6. Make sure `tyCacheChanMaterials` being selected
![image](https://github.com/ynput/ayon-core/assets/64118225/158923b0-f37f-4b16-866a-da97b2ba8c02)

7. Publish
8. Open new scene
9. Load the tyCache instance, there should be instanced materials lib found in current file info, which means the material should be also loaded along with tyCache.
![image](https://github.com/ynput/ayon-core/assets/64118225/a507dede-408b-48eb-a9d8-76b600653308)
